src/Pos/Wallet/Web/Server/Methods.hs

```diff
@@ -586,27 +548,20 @@ getMoneySourceWallet (AddressMoneySource addrId) = cwamWId addrId
 getMoneySourceWallet (AccountMoneySource accId)  = aiWId accId
 getMoneySourceWallet (WalletMoneySource wid)     = wid
 
--- Read-only function.
-computeTxFee
-    :: WalletWebMode m
-    => MoneySource
-    -> NonEmpty (CId Addr, Coin)
-    -> m CCoin
-computeTxFee moneySource dstDistr = mkCCoin <$> do
-    feePolicy <- bvdTxFeePolicy <$> gsAdoptedBVData
-    case feePolicy of
-        TxFeePolicyUnknown w _               -> throwM $ unknownFeePolicy w
-        TxFeePolicyTxSizeLinear linearPolicy -> do
-            txAux <- stabilizeTxFee linearPolicy moneySource dstDistr >>= createFakeTxFromRawTx
-            eitherToThrow invalidFee .
-                integerToCoin .
-                ceiling .
-                calculateTxSizeLinear linearPolicy .
-                biSize @TxAux $ txAux
-  where
-    invalidFee reason = InternalError ("Invalid fee: " <> reason)
-    unknownFeePolicy =
-        InternalError . sformat ("UnknownFeePolicy, tag: "%build)
+getMoneySourceUtxo :: WalletWebMode m => MoneySource -> m Utxo
+getMoneySourceUtxo =
+    getMoneySourceAddresses >=>
+    mapM (decodeCIdOrFail . cwamWId) >=>
+    getOwnUtxos
+
+-- [CSM-237] It should be moved to `Pos.Wallet.Web.Mode`, but
+-- to make it possible all this mess should be neatly separated
+-- to modules and refactored
```

Comment:
P.S: it's because of imports which are really hard to be made non-circular in this situation.